src/Backend.php

```diff
@@ -16,21 +17,21 @@ class Backend
     protected $backendManager;
 
     /**
-     * @param array $config
+     * @param array           $config
+     * @param ClientInterface $client
      */
-    public function __construct($config)
+    public function __construct($config, ClientInterface $client = null)
     {
         $domains = $config['domains'];
         // stay compatible to old configs
         if (isset($config['domain'])) {
             array_push($domains, $config['domain']);
         }
 
-        $clientOptions = [];
-        if (isset($config['client'])) {
-            $clientOptions = $config['client'];
+        if (null === $client) {
```

Comment:
Don't know how the exact CGL of  this project is, but I dislike yoda style.
`if (!$client)`